coordinator/persistence-api/src/main/java/io/stargate/db/schema/SchemaHashable.java

```diff
@@ -24,17 +25,21 @@ public interface SchemaHashable {
 
   static int hashCode(SchemaHashable object) {
     if (object == null) {
-      return 0;
+      return -1;
     }
     return object.schemaHashCode();
   }
 
+  // 12-Apr-2023, tatu: This is still dangerous; should get rid of it. But reduce usage for now
   static int hashCode(Object object) {
     // It's still possible that something like `Optional<Enum>` would make it through. So that's
     // something to look out for.
-    if (object instanceof Enum || object instanceof SchemaHashable) {
+    if (object instanceof Enum
+        || object instanceof SchemaHashable
+        || object instanceof Collection) {
```

Comment:
Preventing calls with Collections (Lists, Sets); would have caught the bug we had.